WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/BloggingReminderTimePicker.kt

```diff
@@ -0,0 +1,66 @@
+package org.wordpress.android.ui.bloggingreminders
+
+import android.app.Dialog
+import android.app.TimePickerDialog
+import android.content.Context
+import android.content.DialogInterface.BUTTON_NEGATIVE
+import android.os.Bundle
+import android.text.format.DateFormat
+import androidx.fragment.app.DialogFragment
+import androidx.lifecycle.ViewModelProvider
+import org.wordpress.android.WordPress
+import java.util.Calendar
+import javax.inject.Inject
+
+class BloggingReminderTimePicker : DialogFragment() {
+    @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
+    private lateinit var viewModel: BloggingRemindersViewModel
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        viewModel = ViewModelProvider(requireActivity(), viewModelFactory).get(BloggingRemindersViewModel::class.java)
+        // Use the current time as the default values for the picker
+        val c = Calendar.getInstance()
+        val hour = c.get(Calendar.HOUR_OF_DAY)
+        val minute = c.get(Calendar.MINUTE)
+        val is24HrFormat = DateFormat.is24HourFormat(activity)
+        val timePicker = TimePickerDialog(
+                activity,
+                { _, selectedHour, selectedMinute ->
+                    viewModel.onChangeTime(selectedHour, selectedMinute)
+                },
+                hour,
+                minute,
+                is24HrFormat)
+
+        timePicker.setCanceledOnTouchOutside(false)
+
+        timePicker.setOnShowListener {
+            timePicker.getButton(BUTTON_NEGATIVE).setOnClickListener {
+                timePicker.dismiss()
+                viewModel.onCancelTimeSelection()
+            }
+        }
+
+        return timePicker
+    }
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        (requireActivity().applicationContext as WordPress).component().inject(this)
+    }
+
+    companion object {
+        const val TAG = "blogging_reminders_time_picker"
+        private const val ARG_HOUR = "arg_hour"
+        private const val ARG_MINUTE = "arg_minute"
+
+        fun newInstance(hour: Int, minute: Int): BloggingReminderTimePicker {
+            return BloggingReminderTimePicker().apply {
+                arguments = Bundle().apply {
+                    putInt(ARG_HOUR, hour)
+                    putInt(ARG_MINUTE, minute)
+                }
+            }
+        }
```

Comment:
It seems that we're not using these arguments anywhere yet. I imagine you're planning to use them to pass the currently selected time to the dialog? If so, I think that's the way to go 👍

Reply:
Yes, once I merge the FluxC PR

Reply:
Removed the arguments now, no longer needed as time and hour are available after FluxC update

Reply:
I believe we should still pass the currently selected time to the dialog. If we don't, the time picker will display a different time from the one that's being displayed on the bottom sheet:


https://user-images.githubusercontent.com/830056/129103685-a9fbb2cd-ab2f-4e20-b06a-1fcd0fea8b4d.mp4

Reply:
Updated to reflect current time selection on the picker